decompress_testing.go

```diff
@@ -47,7 +47,7 @@ func TestDecompressor(t testing.T, d Decompressor, cases []TestDecompressCase) {
 			defer os.RemoveAll(td)
 
 			// Decompress
-			err := d.Decompress(dst, tc.Input, tc.Dir)
+			err := d.Decompress(dst, tc.Input, tc.Dir, 0022)
```

Comment:
It doesn't look like we assert the correct mode is used below. Could you add that?

Reply:
This turns out to be an awkward place to check permissions, I've added a separate test for permission preserving and masking.